tutorials/time-freq/ssvep.py

```diff
@@ -0,0 +1,382 @@
+
+psds, freqs = mne.time_frequency.psd_welch(
+    epochs,
+    n_fft=int(sf * (tmax - tmin)), n_overlap=int(sf * .5), n_per_seg=None,
+    tmin=tmin, tmax=tmax,
+    fmin=fmin, fmax=fmax)
+
+
+###############################################################################
+# Extract SSVEP/vSSR
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#
+# The function below calculates the ratio of power in the target frequency bin
+# to average power in a set of neighbor (noise) bins. The composition of noise
+# bins can be tweaked by two parameters:
+# - how many noise bins do you want?
+# - do you want to skip n bins directly next to the target bin?
+
+###############################################################################
+# SNR calculation function
+# ^^^^^^^^^^^^^^^^^^^^^^^^
+def snr_spectrum(psds, noise_n_neighborfreqs=1, noise_skip_neighborfreqs=1):
+    """
+    Parameters
+    ----------
+    psds - np.array
+        containing psd values as spit out by mne functions. must be 2d or 3d
+        with frequencies in the last dimension
+    noise_n_neighborfreqs - int
+        number of neighboring frequencies used to compute noise level.
+        increment by one to add one frequency bin ON BOTH SIDES
+    noise_skip_neighborfreqs - int
+        set this >=1 if you want to exclude the immediately neighboring
+        frequency bins in noise level calculation
+
+    Returns
+    -------
+    snr - np.array
+        array containing snr for all epochs, channels, frequency bins.
+        NaN for frequencies on the edge, that do not have enoug neighbors on
+        one side to calculate snr
+    """
+
+    # prep not epoched / single channel data
+    is_2d = True if (len(psds.shape) == 2) else False
+    if is_2d:
+        psds = psds.reshape((1, psds.shape[0], psds.shape[1]))
+
+    # SNR loop
+    snr = np.empty(psds.shape)
+    for i_freq in range(psds.shape[2]):
+
+        # skip freqs on the edges (without noise neighbors)
+
+        start_freq_i = noise_n_neighborfreqs + noise_skip_neighborfreqs
+        stop_freq_i = (psds.shape[2] - noise_n_neighborfreqs
+                       - noise_skip_neighborfreqs)
+        if not (stop_freq_i > i_freq >= start_freq_i):
+            snr[:, :, i_freq] = np.nan
+            continue
+
+        # extract signal level
+        signal = psds[:, :, i_freq]
+
+        # ... and average noise level
+        i_noise = []
+        for i in range(noise_n_neighborfreqs):
+            i_noise.append(i_freq + noise_skip_neighborfreqs + i + 1)
+            i_noise.append(i_freq - noise_skip_neighborfreqs - i - 1)
+        noise = psds[:, :, i_noise].mean(axis=2)
+
+        snr[:, :, i_freq] = signal / noise
+
+    # reshape not epoched / single channel data to original dimensions
+    if is_2d:
+        snr = snr.reshape(snr.shape[1], snr.shape[2])
+
+    return snr
+
+
+###############################################################################
+# Calculate SNR
+# ^^^^^^^^^^^^^
+
```

Comment:
here we might add some text that explainds what is actually done to the ppl who read the tutorial.
smth like: 
"now we call the function to compute our snr spectrum. SNR is a relative measure: it's the ratio of power in a given frequency bin compared to a baseline - the average power in the surrounding frequency bins. hence, we need to define some parameters for this 'baseline' - how many neighboring bins should be taken for this computation, and do we want to skip the direct neighbors (this can make sense if the stimulation frequency is not super constant, or frequency bands are very narrow)"

Reply:
Added your text verbatim. Can edit later.